File: exercicios/codigosEmOutrasLinguagens/EX9DIC/EX9DIC.py
def buscaReversa(dic, valor_buscado):
    chaves_encontradas = []
    
    for chave in dic:
        valor = dic.get(chave)
        if valor == valor_buscado:
            chaves_encontradas.append(chave)
    
    return chaves_encontradas

def main():
    dic = {}
    continuar = True

    while continuar:
        chave = input("Digite uma chave para ser armazenada no dicionário (enter para parar): ")
        if chave == "": 
            continuar = False
            break
        
        valor = input("Digite um valor para ser armazenado no dicionário (enter para parar): ")
        
        dic[chave] = valor
        
    valor_buscado = input("\nPor fim, digite um valor a ser buscado: ")
    
    chaves_encontradas = buscaReversa(dic, valor_buscado)
    print("\nChaves que possuem o valor buscado:")
    print(chaves_encontradas)
    
if __name__ == "__main__":
    main()

File: exercicios/codigosEmOutrasLinguagens/EX9DIC/EX9DIC.c
#include <stdio.h>
#include <string.h>

#define MAX_SIZE 100

void buscaReversa(char chaves[MAX_SIZE][MAX_SIZE], char valores[MAX_SIZE][MAX_SIZE], int tamanho, char valor_buscado[MAX_SIZE]) {
    int i;
    printf("\nChaves que possuem o valor buscado '%s':\n", valor_buscado);
    
    for (i = 0; i < tamanho; i++) {
        if (strcmp(valores[i], valor_buscado) == 0) {
            printf("%s\n", chaves[i]);
        }
    }
}

int main() {
    char chaves[MAX_SIZE][MAX_SIZE];
    char valores[MAX_SIZE][MAX_SIZE];
    int tamanho = 0;
    
    char chave[MAX_SIZE], valor[MAX_SIZE];
    
    while (1) {
        printf("Digite uma chave para ser armazenada no dicionário (enter para parar): ");
        fgets(chave, MAX_SIZE, stdin);
        chave[strcspn(chave, "\n")] = 0; 
        
        if (strlen(chave) == 0) break;
        
        printf("Digite um valor para ser armazenado no dicionário: ");
        fgets(valor, MAX_SIZE, stdin);
        valor[strcspn(valor, "\n")] = 0; 
        
        strcpy(chaves[tamanho], chave);
        strcpy(valores[tamanho], valor);
        tamanho++;
    }
    
    char valor_buscado[MAX_SIZE];
    printf("\nPor fim, digite um valor a ser buscado: ");
    fgets(valor_buscado, MAX_SIZE, stdin);
    valor_buscado[strcspn(valor_buscado, "\n")] = 0; 
    
    buscaReversa(chaves, valores, tamanho, valor_buscado);
    
    return 0;
}